clients/src/main/java/org/apache/kafka/common/serialization/ListSerializer.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.serialization;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.utils.Utils;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class ListSerializer<T> implements Serializer<List<T>> {
+
+    private Serializer<T> inner;
+    private Boolean isPrimitive;
+
+    private List<Class> primitiveSerializers = Arrays.asList(
+            LongSerializer.class,
+            IntegerSerializer.class,
+            ShortSerializer.class,
+            FloatSerializer.class,
+            DoubleSerializer.class);
```

Comment:
Shouldn't we also add `BytesSerializer` and `ByteArraySerializer` here?  Same question for deserialization.

+1 to also adding UUID serializer (and deserializer).

Reply:
@mjsax mentioned during the KIP discussion:
```
For the primitive serializers: `BytesSerializer` is not primitive IMHO,
as is it for `byte[]` with variable length -- it's for arrays, not for
single `byte` (note, that `Bytes` is a Kafka class wrapping `byte[]`).
```

I'll add UUID (de)serializers